test_scripts/API/AddSubMenu/ATF_AddSubMenu_Negative_SubMenuIcon_SpaceInType.lua

```diff
@@ -0,0 +1,83 @@
+---------------------------------------------------------------------------------------------
+-- Requirement summary:
+--		[GENIVI] AddSubMenu: SDL must support new "subMenuIcon" parameter 
+--		[GeneralResultCodes] INVALID_DATA wrong characters
+--
+-- Description:
+-- 		Mobile app sends AddSubMenu with "subMenuIcon" that has space in imageType
+-- 1. Used preconditions:
+-- 		Delete files and policy table from previous ignition cycle if any
+-- 		Start SDL and HMI
+--      Activate application
+-- 2. Performed steps:
+-- 		Send AddSubMenu RPC with "subMenuIcon" with space in imageType
+--
+-- Expected result:
+-- 		SDL must respond with INVALID_DATA and "success":"false"
+---------------------------------------------------------------------------------------------
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+config.SDLStoragePath = config.pathToSDL .. "storage/"
+local storagePath = config.SDLStoragePath..config.application1.registerAppInterfaceParams.appID.. "_" .. config.deviceMAC.. "/"
+
+--[[ General configuration parameters ]]
+Test = require('connecttest')	
+require('cardinalities')
+
+--[[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+require('user_modules/AppTypes')
+	
+--[[ Preconditions ]]
+commonSteps:DeleteLogsFileAndPolicyTable()
+commonFunctions:newTestCasesGroup("Preconditions")
+function Test:Precondition_ActivateApp()
+	local RequestId = self.hmiConnection:SendRequest("SDL.ActivateApp", { appID = self.applications["Test Application"]})
+	EXPECT_HMIRESPONSE(RequestId)
+	:Do(function(_,data)
+		if data.result.isSDLAllowed ~= true then
+			local RequestIdGetMes = self.hmiConnection:SendRequest("SDL.GetUserFriendlyMessage", 
+			{language = "EN-US", messageCodes = {"DataConsent"}})
+			EXPECT_HMIRESPONSE(RequestIdGetMes)
+			:Do(function(_,data)
+				self.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality", 
+				{allowed = true, source = "GUI", device = {id = config.deviceMAC, name = "127.0.0.1"}})
+				EXPECT_HMICALL("BasicCommunication.ActivateApp")
+				:Do(function(_,data
```

Comment:
@vvvakulenko,
brackets removed in e7adcf2;
"data" is used below, will not be removed.